sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonParser.scala

```diff
@@ -420,17 +420,17 @@ class JacksonParser(
     case VALUE_STRING if parser.getTextLength < 1 && allowEmptyString =>
       dataType match {
         case FloatType | DoubleType | TimestampType | DateType =>
-          throw QueryExecutionErrors.emptyJsonFieldValueError(dataType)
+          throw EmptyJsonFieldValueException(dataType)
         case _ => null
       }
 
     case VALUE_STRING if parser.getTextLength < 1 =>
-      throw QueryExecutionErrors.emptyJsonFieldValueError(dataType)
+      throw EmptyJsonFieldValueException(dataType)
```

Comment:
the control flow here is bit convoluted, when/where is `EmptyJsonFieldValueException` being handled first?

Reply:
Depending on the JSON object, `EmptyJsonFieldValueException` will be intercepted at convertObject, convertMap, or convertArray (because every JSON value will be treated as either of those three although I am not quite sure about from_json functions)and be wrapped into PartialResultException. This exception will eventually be converted into BadRecordException in `JacksonParser.parse` method and processed in `FailureSafeParser` according to the parsing mode.

Reply:
I hope this answered your question.

Reply:
is it more efficient to return null directly in the field converter, to avoid exception creation and wrapping? I think the implementation should be simple, we return null only when the mode is PERMISSIVE and enablePartitionResult is true.

Reply:
I don't know if this is more efficient because you would have to check the parsing mode before returning a result or throwing an exception. Also, in permissive mode, we still want to capture the exception when badRecordsPath is enabled or _corrupt_record column is provided.

Reply:
oh this is a good point, we should still throw exceptions to not break other features.